src/app/src/components/UtilityGuard.js

```diff
@@ -25,6 +27,15 @@ export default function UtilityGuard({ children }) {
 
 function ContributorUtilityGuard({ children }) {
     const dispatch = useDispatch();
+    const { pathname } = useLocation();
+
+    const boundaryId = pathname.match(/^\/submissions\/(\d+)/)?.[1];
```

Comment:
We use this here instead of `useBoundaryId` because this component is above the `Route` that declares that parameter